### PR TITLE
ts init

### PR DESCRIPTION
**Pull request by AhmedEldessouki**

### CODE REVIEW

File: src/components/Auth/SignIn.tsx [hunk 1]
```diff
@@ -9,18 +9,21 @@ import Layout from '../Layout'
...
     checkUserCredentials,
   ] = useVerifyUserSignInCredentials()
   const {status, dispatch} = useAsync()
 
-  async function submitUserCredentials(e) {
+  async function submitUserCredentials(e: React.SyntheticEvent) {
     e.preventDefault()
     dispatch({type: 'pending'})
-    const {email, password} = e.target.elements
+    const {email, password} = e.target as typeof e.target & {
```
Comment (Aprillion) [inline src/components/Auth/SignIn.tsx:23]: you lost `.elements` ... is that on purpose?

File: src/components/Utils/hooks.tsx [hunk 2]
```diff
@@ -1,15 +1,18 @@
...
-    return () => (mounted.current = false)
+    return () => {
+      mounted.current = false
+    }
   }, [])
 
   return React.useCallback(
-    (...args) => (mounted.current ? dispatch(...args) : void 0),
+    ({...args}): React.Dispatch<IAction> | void =>
+      mounted.current ? dispatch({...args}) : void 0,
```
Comment (Aprillion) [inline src/components/Utils/hooks.tsx:15]: `...args` for https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/rest_parameters is a different feature than the object spead `{...args}` - please don't convert

File: src/components/Auth/SignIn.tsx (hunk 1 above)
Comment (AhmedEldessouki) [reply src/components/Auth/SignIn.tsx:23]: yes, couldn't get it typed. i check the functionality and it was still working.

**Comment by AhmedEldessouki** [comment]
@all-contributors please add @Aprillion for code review

**Pull request merged**